### ISSUE TITLE
ReadWriteLock not working.

### ISSUE THREAD

**Issue by bleujin** [redisson/redisson#1181]
Hello.

I'm interested in this project. I am creating a sample project(https://github.com/bleujin/CrakenRedis) using this, 

**And I do not think readswriteLock is working properly.**

The error pattern is a bit different, but the example org.redission.example.ReadWriteLockExamples.java is not working properly. Version 2.10.5 does not work either.

I do not know much about redis
I hope this project can be maintained.

**Comment by jackygurui** [comment]
Can you explain in a bit more detail as why do you think ReadWriteLock is not working properly?

**Comment by bleujin** [comment]
Thanks for your answer.

Examples related to locking, for example
org.redisson.example.locks.ReadWriteLockExample does not end.

Creating a lock-related sample does not work as expected. For example, unlock does not work right away and it takes time as lock timeout. (Setting timeout to 1 sec and repeating 10 locks and unlocks takes more than 10 seconds.)

I've written a simple TestCase.

```java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.concurrent.Future;
import java.util.concurrent.TimeUnit;
import java.util.concurrent.atomic.AtomicInteger;

import org.redisson.Redisson;
import org.redisson.api.RLock;
import org.redisson.api.RReadWriteLock;
import org.redisson.api.RedissonClient;

import junit.framework.TestCase;

public class TestLock extends TestCase {

	public void testLock() throws Exception {
		RedissonClient redisson = Redisson.create();

		ExecutorService epool = Executors.newFixedThreadPool(10) ;
		AtomicInteger ai = new AtomicInteger() ;
		
		Runnable r = () -> {
			RReadWriteLock rwlock = redisson.getReadWriteLock("lock");
			RLock lock = rwlock.writeLock();
			try {
				lock.lock(1, TimeUnit.SECONDS);
				Thread.sleep(new Random().nextInt(100));
				System.out.print(ai.incrementAndGet()); // do something
			} catch (Exception ex) {
				ex.printStackTrace();
			} finally {
				lock.unlock();
			}
		};
		
		List<Future> list = new ArrayList() ;
		for (int i = 0; i < 10 ; i++) {
			list.add(epool.submit(r)) ;
		}
		
		for (Future f : list) {
			f.get() ;
		}
		
		System.out.println(ai.get()) ;
		redisson.shutdown();
	}

}
```

My specification is redis installed on Windows 10 HomeEdititon and connected to localhost.

**Comment by jackygurui** [comment]
Hi, I can't seem to reproduce the issue with your code.

I am wondering how you have measured the time since your code doesn't not show any metrics and measurements.

**Comment by bleujin** [comment]
I'm sorry I did not have enough explanation.

In the testLock test case
When 10 threads concurrently run within 100ms, the worst case is that the total execution time should be less than 1sec.

However, in practice, it takes 10 seconds over. I think that this is because unlock is not working properly , also that it will not catch a new lock after 100ms(The max release time is 1 second, but the Runnable is done within 100ms.)


By comparison, testLock2 uses java's readWriteLock. This case is terminated within one second as expected. (excluding redis connection time)


Does your computer run testLock in less than a second?

**Comment by bleujin** [comment]
Oh sorry.
I think I found the reason after reading your article again.

I thought Map and Lock would use the same name because the categories are different. However, it does not work if I use the same name.

The first error message was "java.util.concurrent.ExecutionException: java.lang.IllegalMonitorStateException: attempt to unlock lock, not locked by current thread by node id: 1b173326-3eb3-490f-876c-8ce5d38c7682 thread-id: 21" I thought it was limited to concurrency problems.

to sum up
'' '
redisson.getMap ("mymap")
redisson.getReadWriteLock ("mymap")
'' '
I knew I should not use the same name as above.

**Comment by jast90** [comment]
The code as work can work?
```java
public void saveEntityRef(String entityFullName, String referedEntityFullName) {
        Redisson redisson = AppContext.getBean("redisson", Redisson.class);
        RReadWriteLock readWriteLock = ctrmredisson.getReadWriteLock(String.format("%s-%s", entityFullName, referedEntityFullName));
        RLock readLock = readWriteLock.readLock();
        RLock writeLock = readWriteLock.writeLock();
        readLock.lock();
        try {
            boolean exit = ... // query db
            if (exit) {
                return;
            }
        }  finally {
            readLock.unlock();
        }

        writeLock.lock();
        try {
            ...// insert db
        }  finally {
            writeLock.unlock();
        }
    }
```
